src/software/multithreading/observer.h

```diff
@@ -13,7 +13,13 @@ template <typename T>
 class Observer
 {
    public:
-    Observer(size_t buffer_size = DEFAULT_BUFFER_SIZE);
+    /**
+     * Creates a new Observer
+     *
+     * @param buffer_size size of the buffer
+     * @param log_buffer_full whether or not to log when the buffer is full
+     */
+    Observer(size_t buffer_size = DEFAULT_BUFFER_SIZE, bool log_buffer_full = true);
```

Comment:
Isn't this something we would want to know about? 

would it be better to throttle the msg (maybe every second) instead? 

Like 

" buffer full for 2 seconds ... " 
" buffer full for 3 seconds ... "

instead of 
" buffer full "
" buffer full "
" buffer full "
" buffer full "
" buffer full "

Reply:
k

Reply:
That was a question, not a directive

Reply:
We actually expect it always to be full because AI tick time is slightly longer than the vision period and we have a buffer of 1 since we always want to be processing the latest World

Reply:
This would be a nice-to-have for `ThreadedObserver` but pretty out of scope for this PR, maybe open a ticket for it